### PR TITLE
Fix choice set default timeout behavior

### LINKED ISSUE #1891: Fix SDLChoiceSet default timeout
### Bug Report
Currently, it's possible for `SDLChoiceSet` default timeouts to be used incorrectly.

##### Reproduction Steps
1. Create a `SDLChoiceSet` without the timeout set
2. Update the default timeout of `SDLChoiceSet`

##### Expected Behavior
The choice set created before the update uses the new default choice set time

##### Observed Behavior
The choice set uses the old default choice set time.

##### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 7.1.0-develop
* Testing Against: n/a

##### Test Case, Sample Code, and / or Example App
n/a

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #1891

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
Test cases for setting and getting the `SDLChoiceSet`'s `defaultTimeout` and `timeout` properties were added to the **SDLChoiceSetSpec**.

#### Core Tests
Tested with Manticore.

Core version / branch / commit hash / module tested against: Manticore (SDL Core v7.0.0)
HMI name / version / branch / commit hash / module tested against: Manticore (Generic HMI v0.9.0)

### Summary
Fixed the behavior of the`SDLChoiceSet`'s `defaultTimeout` and `timeout` properties:
1. If the `timeout` value is set to `0`, getting the `timeout` value now returns the `defaultTimeout` value instead of `0`.
2. The `timeout` value is now capped to a 5-100 second range. If set to a value less than 5, it will be capped at 5. If set to a value greater than 100, it will be capped to 100. 
1. The `timeout` value can now be set to any value in the `SDLChoiceSet` inits. Previously the object would not be created if the `timeout` was not set to a value in the 5-100 range. 
1. If the `SDLChoiceSet.defaultTimeout` is changed after the `SDLChoiceSet` object is created without the `timeout` value set, the new `SDLChoiceSet.defaultTimeout` is used to set the `timeout` value instead of the old `SDLChoiceSet.defaultTimeout ` value. 

##### Bug Fixes
* Fixed issues with getting the `SDLChoiceSet`'s `defaultTimeout` and `timeout` properties.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/public/SDLChoiceSet.m [hunk 1]
```diff
@@ -57,7 +58,9 @@ - (instancetype)initWithTitle:(NSString *)title delegate:(id<SDLChoiceSetDelegat
         return nil;
     }
 
-    if (timeout < 5 || timeout > 100) {
+    if (timeout == 0) {
+        SDLLogV(@"Creating a choice set with a 0 second timeout; the default timeout will be used: %f seconds", SDLChoiceSet.defaultTimeout);
```
Comment (joeljfischer) [inline SmartDeviceLink/public/SDLChoiceSet.m:62]:
```suggestion
        SDLLogV(@"Creating a choice set with a 0 second timeout; the default timeout will be used: %.01f seconds", SDLChoiceSet.defaultTimeout);
```

This probably needs to reduce the number of decimals.

File: SmartDeviceLink/public/SDLChoiceSet.m [hunk 2]
```diff
@@ -57,7 +58,9 @@ - (instancetype)initWithTitle:(NSString *)title delegate:(id<SDLChoiceSetDelegat
...
+    } else if (timeout < 5 || timeout > 100) {
```
Comment (joeljfischer) [inline SmartDeviceLink/public/SDLChoiceSet.m:63]: This isn't capped this way in `AlertView`. We want the same system across both.

File: SmartDeviceLink/public/SDLChoiceSet.m [hunk 3]
```diff
@@ -144,6 +147,14 @@ - (void)setHelpList:(nullable NSArray<SDLVRHelpItem *> *)helpList {
     }
 }
 
+- (NSTimeInterval)timeout {
+    if (_timeout == TimeoutDefault) {
+        return SDLChoiceSet.defaultTimeout;
+    }
+
+    return _timeout;
+}
```
Comment (joeljfischer) [inline SmartDeviceLink/public/SDLChoiceSet.m:156]: This doesn't look the same as `SDLAlertView` (neither does `+(NSTimeInterval)defaultTimeout`

**Review by joeljfischer — CHANGES_REQUESTED** [review]

File: SmartDeviceLink/public/SDLChoiceSet.m (hunk 1 above)
Comment (NicoleYarroch) [reply SmartDeviceLink/public/SDLChoiceSet.m]: This log was removed

File: SmartDeviceLink/public/SDLChoiceSet.m (hunk 2 above)
Comment (NicoleYarroch) [reply SmartDeviceLink/public/SDLChoiceSet.m]: Updated to cap the timeout range between 5 and 100 seconds.

File: SmartDeviceLink/public/SDLChoiceSet.m (hunk 3 above)
Comment (NicoleYarroch) [reply SmartDeviceLink/public/SDLChoiceSet.m]: Updated to cap the timeout range between 5 and 100 seconds.

**Review by joeljfischer — APPROVED** [review]
This will need to be aligned before merge

**Pull request merged**